### PR TITLE
feat: adds fundamentals 4 introduction

### LINKED ISSUE #25977: Foundations/Fundamentals 4: Add Introduction section
### Describe your suggestion

The lesson needs an Introduction section added

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-fundamentals-part-4

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by cakegod**
## This PR
Adds an introduction to fundamentals 4 in foundations path.

## Issue
Closes #25977

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by cakegod** [comment]
On hold for further changes.

File: foundations/javascript_basics/fundamentals-4.md
Comment (CBID2) [inline foundations/javascript_basics/fundamentals-4.md:5]: [Proofreading] This paragraph is off to a great start. It just needed a bit of rewording to make it clear. :) 
```suggestion
Luckily, there is a better way to handle this problem. In this lesson, you’ll learn about arrays, which are data structures that can store multiple values in a single variable. They are very useful for organizing and manipulating large amounts of data. You’ll also learn about loops, control structures that allow you to execute a block of code repeatedly. Loops are very handy for performing the same operation on each element of an array. Finally, you’ll be introduced to Test-Driven Development (TDD). It's where you create tests for your code before writing it.
```

File: foundations/javascript_basics/fundamentals-4.md
Comment (CBID2) [inline foundations/javascript_basics/fundamentals-4.md:3]: [Proofreading] This paragraph is off to a great start. It just needed a bit of rewording to make it clear. :) 
```suggestion
One of the challenges of programming is dealing with large amounts of data. For example, if you want to store the names of all the students in your class, how would you do it? You could create a variable for each name, but that would be tedious and inefficient. It'll also be hard to manage and update. What if you want to change or access the names later?
```

**Review by CBID2 — COMMENTED** [review]
Hi @cakegod ! :) I pointed out a few proofreading errors while reviewing your PR, so check out my inline comments to help you revise them. In the future, consider running your suggestions through Grammarly or Spellcheck before making a pull request. Other than that, you're off to a great start! :)
*Note: I'm not a maintainer. Just a friendly contributor who wants to help other contributors. :)

File: foundations/javascript_basics/fundamentals-4.md
Comment (cakegod) [reply foundations/javascript_basics/fundamentals-4.md:5]: I generally like those changes; the repetition of 'array' was done purposefully, but I can see it might not be necessary and maybe counterproductive. As for the last phrase, I would like to clarify that it's a _practice_ of writing tests.

File: foundations/javascript_basics/fundamentals-4.md
Comment (cakegod) [reply foundations/javascript_basics/fundamentals-4.md:3]: Good changes.

File: foundations/javascript_basics/fundamentals-4.md
Comment (CBID2) [reply foundations/javascript_basics/fundamentals-4.md:5]: > I generally like those changes; the repetition of 'array' was done purposefully, but I can see it might not be necessary and maybe counterproductive. As for the last phrase, I would like to clarify that it's a _practice_ of writing tests.

Ok

Comment (cakegod) [reply foundations/javascript_basics/fundamentals-4.md:5]: Thanks for your help!

File: foundations/javascript_basics/fundamentals-4.md
Comment (ManonLef) [inline foundations/javascript_basics/fundamentals-4.md:5]:
```suggestion
Luckily, there is a better way to handle this problem. In this lesson, you’ll learn about arrays, which are data structures that can store multiple values in a single variable. Arrays are very useful for organizing and manipulating large amounts of data. You’ll also learn about loops, which are control structures that allow you to execute a block of code repeatedly. Loops are very handy for performing the same operation on each element of an array. Finally, you’ll be introduced to Test-Driven Development (TDD), which is the practice of writing tests for your code before you write the code itself.
```

Wonderful job Cake! Considering learners are very new, I reverted back to being a bit more descriptive and repetitive as well. I think being extra clear what is meant with these new terms is crucial.

File: foundations/javascript_basics/fundamentals-4.md
Comment (ManonLef) [reply foundations/javascript_basics/fundamentals-4.md]: Thanks for providing feedback Christine. 

I have to agree with Cake here that the repetition of array is a good idea since it's the first time people hear about it. 

I also left feedback on the last sentence. Another newly introduced concept in the curriculum

File: foundations/javascript_basics/fundamentals-4.md
Comment (cakegod) [reply foundations/javascript_basics/fundamentals-4.md:5]: I don't have strong feelings about either, so I'll follow your feedback.

**Review by zachmmeyer — APPROVED** [review]
Thanks everyone!

**Pull request merged**
